src/iperf_api.c

```diff
@@ -4889,16 +4889,7 @@ iperf_json_finish(struct iperf_test *test)
         cJSON_Delete(test->json_top);
         test->json_top = NULL;
     }
-    // Get ASCII rendering of JSON structure.  Then make our
-    // own copy of it and return the storage that cJSON allocated
-    // on our behalf.  We keep our own copy around.
-    char *str = cJSON_Print(test->json_top);
-    if (str == NULL)
-	return -1;
-    test->json_output_string = strdup(str);
-    cJSON_free(str);
-    if (test->json_output_string == NULL)
-        return -1;
+
```

Comment:
This duplicated code, which appears to have come when we merged in PR #1098, is responsible for the error exit (because `test->json_top` may have been cleared in line 4890). So deleting this is good.